code_search/kubeflow/components/search-index-creator.jsonnet

```diff
@@ -3,5 +3,82 @@ local nms = import "nms.libsonnet";
 
 local env = std.extVar("__ksonnet/environments");
 local params = std.extVar("__ksonnet/params").components["search-index-creator"];
+local experiments = import "experiments.libsonnet";
 
-std.prune(k.core.v1.list.new(nms.parts(params, env).creator))
+local baseParams = std.extVar("__ksonnet/params").components["submit-code-embeddings-job"];
+local experimentName = baseParams.experiment;
+local params = baseParams + experiments[experimentName] + {
+  name: experimentName + "-create-search-index",
+};
+
+local jobSpec = {
+  apiVersion: "batch/v1",
+  kind: "Job",
+  metadata: {
+    name: params.name,
+    namespace: env.namespace,
+    labels: {
+      app: params.name,
+    },
+  },
+  spec: {
+    replicas: 1,
+    template: {
+      metadata: {
+        labels: {
+          app: params.name,
+        },
+      },
+      spec: {
+        // Don't restart because all the job should do is launch the Dataflow job.
+        restartPolicy: "Never",
+        containers: [
+          {
+            name: "dataflow",
+            image: params.image,
+            command: [
+		          "python",
+		          "-m",
+		          "code_search.nmslib.cli.create_search_index",
+		          "--data_dir=" + params.dataDir,
+		          "--lookup_file=" + params.lookupFile,
+		          "--index_file=" + params.indexFile,
+		        ],
+            env: [
+              {
+                name: "GOOGLE_APPLICATION_CREDENTIALS",
+                value: "/secret/gcp-credentials/user-gcp-sa.json",
+              },
+            ],
+            // Creating the index requires a lot of memory.
+            resources: {
```

Comment:
not sure if i follow here. the actual workload is send to dataflow. why the job's pod itself need much resource?